connect/src/com/telenor/mobileconnect/ui/OperatorSelectionActivity.java

```diff
@@ -0,0 +1,108 @@
+
+    static final String MCC_MNC_PARAM_NAME = "mcc_mnc";
+    static final String MCC_MNC_PARAM_SEP = "_";
+    static final String SUBSCRIBER_ID_QUERY_PARAM_NAME = "subscriber_id";
+
+    @SuppressLint("MissingSuperCall")
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState, OperatorSelectionWebFragment.class);
+    }
+
+    @Override
+    public void onSuccess(Object successData) {
+
+        Uri uri = (Uri) successData;
+        String[] mccAndMnc = uri.getQueryParameter(MCC_MNC_PARAM_NAME).split(MCC_MNC_PARAM_SEP);
+
+        @SuppressWarnings("unchecked")
+        Map<String, String> parameters = (Map<String, String>) this
+                .getIntent()
+                .getExtras()
+                .get(ConnectUtils.PARAM_SNAPHOT_EXTRA);
+        @SuppressWarnings("unchecked")
+        List<String> uiLocales = (List<String>) this
+                .getIntent()
+                .getExtras()
+                .get(ConnectUtils.UI_LOCALES_SNAPHOT_EXTRA);
+        final int requestCode = this
+                .getIntent()
+                .getExtras()
+                .getInt(ConnectUtils.REQUEST_CODE_EXTRA);
+        final int customLoadingLayout = this
+                .getIntent()
+                .getExtras()
+                .getInt(ConnectUtils.CUSTOM_LOADING_SCREEN_EXTRA, ConnectSdk.NO_CUSTOM_LAYOUT);
+        MobileConnectSdkProfile sdkProfile = (MobileConnectSdkProfile) ConnectSdk.getSdkProfile();
+        sdkProfile.onStartAuthorization(
+                parameters,
+                uiLocales,
+                mccAndMnc[0],
+                mccAndMnc[1],
+                uri.getQueryParameter(SUBSCRIBER_ID_QUERY_PARAM_NAME),
+                new SdkProfile.OnStartAuthorizationCallback() {
+                    @Override
+                    public void onSuccess(Intent nextIntent) {
+                        if (!isConnectActivity(nextIntent)) {
+                            throw new RuntimeException("Wrong intent.");
+                        }
+                        Activity activity = OperatorSelectionActivity.this;
+                        if (customLoadingLayout != ConnectSdk.NO_CUSTOM_LAYOUT) {
+                            nextIntent.putExtra(
+                                    ConnectUtils.CUSTOM_LOADING_SCREEN_EXTRA,
+                                    customLoadingLayout);
+                        }
+                        activity.startActivityForResult(nextIntent, requestCode);
+                    }
+
+                    @Override
+                    public void onError() {
+                        cancelAndShowErrorMessage(OperatorSelectionActivity.this);
+                    }
+                });
+    }
+
+
+
+    @Override
+    public void onError(Object errorData) {
+        cancelAndShowErrorMessage(this);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
```

Comment:
I have skimmed through your PR (https://github.com/telenordigital/connect-android-sdk/pull/57 ):

I think it's OK, at least for not being an "official" PR, but the main intention of my PR was to keep the old Connect Id code as much unchanged as possible (and with as little "glue" needed to reconcile Connect Id and MC as possible), in order to enable the Connect Id team to maintain their stuff without worrying too much about MC. In other words, the plan of my PR was to implement MC merely as a "client" of Connect Id, since both profiles do exactly the same thing after a certain point in the flow has been reached. This plan is kind of broken in your PR, since ConnectActivity and ConnectSdkProfile (and possibly some other classes) implement stuff solely motivated by MC.

I think the design patterns used in both PRs are equally legitimate. This PR implements a variant of delegate pattern, while yours implements something I cannot put a naming label on, but it's obvious that it's OK. The main difference is that my PR tries to keep the differences between Connect Id and MC in the com.telenordigital.mobileconnect package, while yours doesn't impose such restrictions upon itself.

I would have had some more pointed comments if this PR had been "official". For example, I didn't like the sudden "publicness" of ConnectSdk.getAuthorizeUriAndSetLastAuthState(). Similarly, I didn't like setValue() and getValue in SdkProfle interface neither, since these two methods are obviously motivated by some implementation needs, not by the need for establishing a valid contract. Nevertheless, since your PR is still a rough and a non-official one, I will refrain from reviewing it in details.